modules/virt-preparing-container-disk-for-vms.adoc

```diff
@@ -0,0 +1,51 @@
+// Module included in the following assemblies:
+//
+// * virt/virtual_machines/virt-create-vms.adoc
+
+[id="virt-preparing-container-disk-for-vms_{context}"]
+= Preparing a container disk for virtual machines
+
+A container disk must be built with a virtual machine image and pushed
+to a container registry before it can be attached to a virtual machine as a containerDisk volume.
+
+.Prerequisites
+
+* You have root or sudo privileges on your local system to install podman.
+
+* The virtual machine image must be either QCOW2 or RAW format.
+
+.Procedure
+
+. Install `podman` if it is not already installed:
++
+[source,terminal]
+----
+$ sudo yum install podman -y
+----
+
+. Create a Dockerfile to build the virtual machine image into a container.
+The virtual machine image must be owned by QEMU, which has a UID of `107`, and placed in the `/disk` directory inside the container.
++
+[source,terminal]
+----
+$ cat > Dockerfile << EOF
+FROM scratch
+ADD --chown=107:107 <vm_image>.qcow2 /disk/ <1>
```

Comment:
@davidvossel when I tried this, I got myself an containerDisk; but running it produced

```
failed to get image info: failed to invoke qemu-img: exit status 1: 'qemu-img: Could not open '/var/run/kubevirt/container-disks/disk_0.img': Could not open '/var/run/kubevirt/container-disks/disk_0.img': Permission denied '
```

can you tell why? How can I chmod the ADDed file to 0444 in the Dockerfile?

Reply:
This command `ADD --chown=107:107 disk.qcow2 /disk/` worked without issue for me. I'm not sure what you're experiencing here.

Reply:
i think you're correct that we should be setting 444 on disk now. This technically works without 444 if we get the ownership correct, but that's because the ownership allows us to modify the permissions to 444 after the fact.

The only way I'm aware of to set 444 permissions is to do it external of the scratch container's Dockerfile. ADD only has chown built in and the scratch container itself doesn't have the chmod tool internally. 

So if i set 444 permissions on the file before running the `docker build` or `podman build` then it that works.

Reply:
> i think you're correct that we should be setting 444 on disk now. This technically works without 444 if we get the ownership correct, but that's because the ownership allows us to modify the permissions to 444 after the fact.

forget what I said there. 444 shouldn't be necessary. simply setting the owner to 107:107 and ensuring that the file has read permissions for the owner user is the minimum we need right now. The mutation of the permissions to 444 at runtime is something that we shouldn't be doing and will get addressed. that doesn't impact us here though.

Reply:
@davidvossel I don't understand the bottom line. The the suggested `Dockerfile` did not produce a working container for me. What should be done to it to make it work?

Reply:
@davidvossel what is needed to make the CD work?

Reply:
> @davidvossel what is needed to make the CD work?

Our testsuite verifies 444 mode and 107:107 ownership for a virtualmachine image in the disk/ dir of a container. That's what we verify.